File: examples/typescript/02_upload_and_run.ts
import { existsSync } from 'node:fs'
import { readFile } from 'node:fs/promises'
import { basename, dirname, isAbsolute, resolve } from 'node:path'
import { fileURLToPath } from 'node:url'

import { DogeClient } from 'doge-sdk'

const client = new DogeClient({
  baseUrl: process.env.DOGE_DAEMON_URL ?? 'http://127.0.0.1:8901',
  apiToken: process.env.DOGE_API_TOKEN,
})

const scriptDir = dirname(fileURLToPath(import.meta.url))

function sampleDocPath(): string {
  const raw = process.env.DOGE_SAMPLE_DOC
  if (!raw) return resolve(scriptDir, '../../README.md')
  if (isAbsolute(raw) || existsSync(raw)) return raw
  const scriptRelative = resolve(scriptDir, raw)
  if (existsSync(scriptRelative)) return scriptRelative
  const repoRelative = resolve(scriptDir, '../..', raw)
  if (existsSync(repoRelative)) return repoRelative
  return raw
}

const source = sampleDocPath()
const file = new Blob([await readFile(source)], { type: 'text/markdown' })
const document = await client.documents.upload(file, basename(source))
const session = await client.sessions.create('Document-backed research')
const runId = await session.run('Summarize the uploaded document and identify evidence gaps.', {
  document_ids: [document.document_id],
  workflow: 'investment_research',
})

console.log(`run_id=${runId}`)

File: examples/python/02_upload_and_run.py
"""Upload a text document and start a research run."""

from __future__ import annotations

import os
from pathlib import Path

from doge_sdk import DogeClient


def sample_doc_path() -> Path:
    raw = os.getenv("DOGE_SAMPLE_DOC")
    if not raw:
        return Path(__file__).resolve().parents[2] / "README.md"
    candidate = Path(raw).expanduser()
    if candidate.is_absolute() or candidate.exists():
        return candidate
    script_relative = Path(__file__).resolve().parent / candidate
    if script_relative.exists():
        return script_relative
    repo_relative = Path(__file__).resolve().parents[2] / candidate
    if repo_relative.exists():
        return repo_relative
    return candidate


def main() -> None:
    client = DogeClient(
        base_url=os.getenv("DOGE_DAEMON_URL", "http://127.0.0.1:8901"),
        api_token=os.getenv("DOGE_API_TOKEN"),
    )
    try:
        source = sample_doc_path()
        document = client.documents.upload_path(str(source), content_type="text/markdown")
        session = client.sessions.create("Document-backed research")
        run_id = session.run(
            "Summarize the uploaded document and identify evidence gaps.",
            document_ids=[document["document_id"]],
            workflow="investment_research",
        )
        print(f"run_id={run_id}")
    finally:
        client.close()


if __name__ == "__main__":
    main()
